src/graphapi/operationgraph_descriptor.cpp

```diff
@@ -0,0 +1,237 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/errors.hpp>
+#include <miopen/graphapi/operationgraph_descriptor.hpp>
+
+#include <algorithm>
+
+namespace miopen {
+
+namespace graphapi {
+
+OperationGraph::OperationGraph(miopenHandle_t handle, const OpGraph& opGraph) : mHandle(handle)
+{
+    /* TODO: to be implemented in
+     * [MHA] Implement MIOPEN_BACKEND_OPERATIONGRAPH_DESCRIPTOR read only attributes #2914
+     * https://github.com/ROCm/MIOpen/issues/2914
+     */
+    (void)handle;
+    (void)opGraph;
+}
+
+OperationGraph::OperationGraph(miopenHandle_t handle, OpGraph&& opGraph) : mHandle(handle)
```

Comment:
why not just define it as `default` and let the compiler generate it for us?

Reply:
it won't use `handle` argument

Reply:
And this ctor will be developed further by Bibek

Reply:
Why wait for Bibek (@bghimireamd ) when we can just set it to `default` and call it a day? What special operation needs to happen here? The `OpGraph` has a well-defined `move` constructor.

Reply:
If we are going to combine OpGraph and OperationGraph, then this code will be redesinged.